library/common-library/templates/_agent-config.tpl

```diff
@@ -5,13 +5,26 @@ This helper should return the defaults that all agents should have
 {{- if include "common.verboseLog" . }}
 verbose: 1
 {{- end }}
+
 {{- if (include "common.nrStaging" . ) }}
 staging: true
 {{- end }}
+
+{{- with include "common.license._licenseKey" . }}
+{{- /* We will mostly override this using a environment variable but we can have it here just in case */}}
+license: {{ . }}
+{{- end }}
+
 {{- with include "common.proxy" . }}
 proxy: {{ . | quote }}
 {{- end }}
+
 {{- with include "common.fedramp.enabled" . }}
 fedramp: {{ . }}
 {{- end }}
+
+{{- with include "common.customAttributes" . }}
+custom_attributes:
+  {{- . | nindent 2 }}
+{{- end }}
```

Comment:
Are customAttributes allowed to be formatted as YAML? I though they need to be a JSON-encoded string.

Reply:
It is needed as a string if you use it as `NRIA_CUSTOM_ATTRIBUTES` environment variable but you can put a map directly in the YAML: https://github.com/newrelic/infrastructure-agent/blob/master/pkg/config/config.go#L187

Reply:
It seems like they can be YAML: https://github.com/newrelic/infrastructure-agent/blob/master/pkg/config/config.go#L182=